src/math/error.cpp

```diff
@@ -82,23 +95,30 @@ double error(ErrorType errorType, const std::vector<double> &vecA, const std::ve
  */
 
 // Return RMSE between supplied data
-double rmse(const Data1D &A, const Data1D &B, bool quiet)
+double rmse(const Data1D &A, const Data1D &B, bool quiet, Range range)
 {
     // First, generate interpolation of data B
     Interpolator interpolatedB(B);
 
+    // if range is unset
+    if (range.maximum() == range.minimum())
+    {
+        range.set(B.xAxis().front(), B.xAxis().back());
+    }
```

Comment:
What I suggest we do here, given our `std::optional<Range>` argument, is establish the relevant limits in some temporary vars, and check those vars in the main loop:
```suggestion
    // Set calculation range
    auto rangeMin = range ? range.minimum() : B.xAxis().front();
    auto rangeMax = range ? range.maximum() : B.xAxis().back();
```